dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/models/Sample.java

```diff
@@ -76,31 +91,31 @@ public boolean equals(Object o) {
 
         Sample sample = (Sample) o;
 
-        if (!id.equals(sample.id)) return false;
-        if (sex != null ? !sex.equals(sample.sex) : sample.sex != null) return false;
+        if (!batch.equals(sample.batch)) return false;
+        if (!name.equals(sample.name)) return false;
+        if (sex != sample.sex) return false;
```

Comment:
sex was nullable in the previous implementation of this method. Is the sex always included in dbSNP?

Reply:
Interestingly it was nullable in this method, but the precondition in the constructor was already there. I don't think dbSNP enforces a sex to be specified.

Reply:
Like we are not forcing to be not null in the constructor, dbDSNP seems to allow empty sex strings, and the hashCode() method "allows" null sex values, then it we should leave it in this method as it was to avoid possible NPE: 
if (sex != null ? !sex.equals(sample.sex) : sample.sex != null) return false;